### PR TITLE
Adds juice to picking up stuff

### PR DESCRIPTION
**Pull request by Iamgoofball**
Stolen completely from https://github.com/OracleStation/OracleStation/pull/1072
Cool demo webm!
https://file.house/Lc_0.webm

Looks nice, sexy animation. People might whine about "stealth stealing" or tactical sprites but most objects have an inhand anyways, which kind of blows the whistle on you grabbing them anyways.

### CODE REVIEW

**Comment by fludd12** [comment]
Ooh, I love this...! Would it be possible to do the reverse, too - make it play a similar animation when putting something down?

**Comment by Iamgoofball** [comment]
Probably.

**Comment by XDTM** [comment]
Just wondering; with the delay, what happens if someone starts picking it up and moves away before it finishes?

**Comment by Iamgoofball** [comment]
i did that a few time sin the video

**Comment by MMMiracles** [comment]
anything that adds more visual context for people who actually watch their surroundings is good

**Comment by MrDoomBringer** [comment]
@Iamgoofball what no you didnt

**Comment by Mickyan** [comment]
why not add the attack animation from the same PR while you're at it

**Comment by FlattestGuitar** [comment]
a) I put this in the wrong file; the animation code should probably be moved to items.dm
b) I had issues with taking things out of bags; the item appeared under the spessman's feet instead of on the backpack, might wanna tweak that, because it looks way better if it's on the bag.

:heart:

**Comment by epochayur** [comment]
will there be any funkiness if someone tries to pick up the item while it's still animating from someone else also picking it up?

**Comment by FlattestGuitar** [comment]
The animation itself is just a copied image of the item. Nothing happens if you click it or walk away from it.

**Comment by Time-Green** [comment]
Make it create a tear in the fabric of reality

**Comment by spoonitate** [comment]
heyo, I helped Flat with this idea! glad to see it's stylish enough for use on other servers.

also, background track is ["Reasoning" from the Persona 4 OST](https://www.youtube.com/watch?v=tHIQdXyNxwY).

File: code/game/atoms_movable.dm [hunk 1]
```diff
@@ -771,3 +771,38 @@
...
+
+	if(!direction)
+		to_y = 16
+
+	flick_overlay(I, GLOB.clients, 6)
+	var/matrix/M = new
+	M.Turn(pick(-30, 30))
+
+	animate(I, transform = M, time = 1)
+	sleep(1)
```
Comment (optimumtact) [inline code/game/atoms_movable.dm]: :clap: sanity :clap: checks :clap: for :clap: item :clap: still :clap: existing :clap: these :clap: are :clap: sleeps :clap:

Comment (optimumtact) [reply code/game/atoms_movable.dm]: also if you're setting wait_for = false these sleeps are literally doing nothing anyway?

**Comment by Borisvanmemes** [comment]
Getting used to these will be hard, if they get merged

**Comment by optimumtact** [comment]
why

**Comment by Grazyn** [comment]
Now I wish picking up items had a sound effect

**Comment by kevinz000** [comment]
:+1: as long as this doesn't actually make you have to stand still or stay in range or something, which according to the code I believe it doesn't.

File: code/game/atoms_movable.dm (hunk 1 above)
Comment (kevinz000) [reply code/game/atoms_movable.dm]: think the waitfor is so it doesn't delay/impact the actual pickup

Comment (Iamgoofball) [reply code/game/atoms_movable.dm]: the sleep is for animationn work

**Comment by JJRcop** [comment]
Iamgoofball/-tg-station#16
clickystuff

**Comment by Iamgoofball** [comment]
yeah the item is already in your inventory as soon as the click happens, the animation is all cosmetic

**Comment by Iamgoofball** [comment]
That's too bad.

**Comment by nicbn** [comment]
Pls remove the shaking

**Comment by Iamgoofball** [comment]
No. Get over it.

**Comment by ExcessiveUseOfCobblestone** [comment]
going against the player's wishes I see, smh!

**Comment by MMMiracles** [comment]
the players are WRONG

**Comment by AlexTheSysop42** [comment]
@Iamgoofball inhands only matter if it's **in** your **hand**, they dont tip anyone off to stealth stealing if you press e fast enough

**Comment by AnturK** [comment]
This is completely broken, @optimumtact read what you merge.

**Comment by Iamgoofball** [comment]
fucking

@JJRcop did you test your PR

**Comment by JJRcop** [comment]
I totally did test yes

**Pull request merged**